auslib/db.py

```diff
@@ -2403,7 +2411,23 @@ def getUserRoles(self, username, transaction=None):
 
     def getAllRoles(self, transaction=None):
         res = self.user_roles.select(columns=[self.user_roles.role], distinct=True, transaction=transaction)
-        return [r["role"] for r in res]
+        roles_list = list(set([r['role'] for r in res]))
```

Comment:
This change to getAllRoles seems unnecessary. getAllUsers already returns all users and the Roles they hold. This view returns exactly the same data, except it makes role the top level key. Are you sure you need this?